nbconvert/filters/markdown_mistune.py

```diff
@@ -29,6 +35,31 @@ class MathInlineGrammar(mistune.InlineGrammar):
     text = re.compile(r'^[\s\S]+?(?=[\\<!\[_*`~$]|https?://| {2,}\n|$)')
 
 
+class MathBlockLexer(mistune.BlockLexer):
+    default_rules = (['block_math', 'latex_environment']
+                     + mistune.BlockLexer.default_rules)
+
+    def __init__(self, rules=None, **kwargs):
+        if rules is None:
+            rules = MathBlockGrammar()
+        super(MathBlockLexer, self).__init__(rules, **kwargs)
+
+    def parse_block_math(self, m):
+        """Add token for a $$math$$ block"""
+        self.tokens.append({
+            'type': 'block_math',
+            'text': m.group(1) or m.group(2)
```

Comment:
This will need to include the possibility of the latex_environment style multiline_math object and so will need to be:
```python
    'text': m.group(1) or m.group(2) or m.group(3)
```